addons/point_of_sale/static/src/js/models.js

```diff
@@ -862,13 +866,34 @@ exports.PosModel = Backbone.Model.extend({
             this.db.add_order(order.export_as_JSON());
         }
 
-        return new Promise(function (resolve, reject) {
-            self.flush_mutex.exec(function () {
-                var flushed = self._flush_orders(self.db.get_orders(), opts);
-
-                flushed.then(resolve, reject);
+        this.set_synch('connecting', self.db.get_orders().length);
 
-                return flushed;
+        return new Promise(function (resolve, reject) {
+            var not_syncing_orders = self.db.get_orders().filter(function (order) {
+                return !order.syncing;
+            });
+            var chunk_size = self.sync_chunk_size; // 2 by default, syncing 2 at a time.
+            var n = Math.ceil(not_syncing_orders.length / chunk_size);
+            var chunked_orders = [];
+            for (var i=0; i < n; i++) {
+                chunked_orders.push(not_syncing_orders.slice(i*chunk_size, (i+1)*chunk_size));
+            }
+            if (chunked_orders.length == 0) {
+                // necessary so that flush_mutex is called at least once.
+                // It is important to call flush_mutex.exec once so that
+                // the chain that this promise returns is not broken.
+                // e.g. in closing the session.
+                chunked_orders.push([]);
+            }
+            chunked_orders.forEach(function (orders) {
```

Comment:
I am not sure you need to add a field syncing on the orders, as when a _flush order succeeds it removes the order from the db. I think  you can make a loop that will get_orders, take the 2 firsts, and then get_orders again while there are orders and no errors from flush.